pkg/apiserver/admission/initializer/types.go

```diff
@@ -22,36 +22,22 @@ import (
 	kubeinformers "k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 
-	gardencoreclientset "github.com/gardener/gardener/pkg/client/core/clientset/internalversion"
-	gardencoreversionedclientset "github.com/gardener/gardener/pkg/client/core/clientset/versioned"
-	gardencoreexternalinformers "github.com/gardener/gardener/pkg/client/core/informers/externalversions"
-	gardencoreinformers "github.com/gardener/gardener/pkg/client/core/informers/internalversion"
+	gardencoreclientset "github.com/gardener/gardener/pkg/client/core/clientset/versioned"
+	gardencoreinformers "github.com/gardener/gardener/pkg/client/core/informers/externalversions"
 	seedmanagementclientset "github.com/gardener/gardener/pkg/client/seedmanagement/clientset/versioned"
 	seedmanagementinformers "github.com/gardener/gardener/pkg/client/seedmanagement/informers/externalversions"
 	settingsinformers "github.com/gardener/gardener/pkg/client/settings/informers/externalversions"
 )
 
-// WantsInternalCoreInformerFactory defines a function which sets InformerFactory for admission plugins that need it.
-type WantsInternalCoreInformerFactory interface {
-	SetInternalCoreInformerFactory(gardencoreinformers.SharedInformerFactory)
-	admission.InitializationValidator
-}
-
-// WantsInternalCoreClientset defines a function which sets Core Clientset for admission plugins that need it.
-type WantsInternalCoreClientset interface {
-	SetInternalCoreClientset(gardencoreclientset.Interface)
-	admission.InitializationValidator
-}
-
 // WantsExternalCoreInformerFactory defines a function which sets external Core InformerFactory for admission plugins that need it.
 type WantsExternalCoreInformerFactory interface {
-	SetExternalCoreInformerFactory(gardencoreexternalinformers.SharedInformerFactory)
+	SetExternalCoreInformerFactory(gardencoreinformers.SharedInformerFactory)
 	admission.InitializationValidator
 }
 
-// WantsExternalCoreClientset defines a function which sets external Core Clientset for admission plugins that need it.
-type WantsExternalCoreClientset interface {
-	SetExternalCoreClientset(gardencoreversionedclientset.Interface)
+// WantsExternalCoreClientSet defines a function which sets external Core Clientset for admission plugins that need it.
```

Comment:
Do we still need to call them as External? We only have one now. What does Kubernetes do?

Reply:
On kubernetes side I see them using `ExternalCoreCLientSet` 

https://github.com/kubernetes/kubernetes/blob/feb18b3f5f9d443c27dd8cccb6358f271f887744/staging/src/k8s.io/apiserver/pkg/admission/initializer/interfaces.go#L32-L35

Reply:
Okay, then let's keep it as it is.

Reply:
I wanted to ask the same and then saw this comment. I don't know why the Kubernetes code wasn't adjusted but in our code it's called `WantsKubeClientset` as well (instead of `WantsExternalKubeClientSet`).

Hence, I would still vote dropping the "external" term here. We also don't have `WantsExternalSeedManagementClientSet` or similar.